src/robot/variables/replacer.py

```diff
@@ -137,46 +137,59 @@ def _get_variable_item(self, match, value):
             logger.warn("Accessing variable items using '%s' syntax "
                         "is deprecated. Use '$%s' instead." % (var, var[1:]))
         for item in match.items:
-            if is_dict_like(value):
-                value = self._get_dict_variable_item(name, value, item)
-            elif is_list_like(value):
-                value = self._get_list_variable_item(name, value, item)
+            if is_sequence(value):
+                value = self._get_sequence_variable_item(name, value, item)
+            elif is_subscriptable(value):
+                value = self._get_subscriptable_variable_item(name, value, item)
```

Comment:
Right, but after changing to the simplified implementation in 2afca766be3f4a2058f4225895890babe4999c31 custom classes that don't inherit from `Sequence` are no longer subscriptable by index/slice (as string). This is a problem for class `ExecutionErrors`, for example. Acceptance tests fail because of this. I am in favor of reverting the aforementioned commit and  tweaking error handling. What do you think?

Reply:
Done this in d7110621c16a371e72b541b70f01cbea7f0f4c8e